packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Deputy/JsiiTypeAttributeBase.cs

```diff
@@ -17,27 +23,32 @@ protected JsiiTypeAttributeBase(Type nativeType, string fullyQualifiedName)
             Load(nativeType.Assembly);
         }
 
-        void Load(Assembly assembly)
+        private static void Load(Assembly assembly)
         {
-            IEnumerable<Assembly> dependencies = assembly.GetReferencedAssemblies()
-                .Select(assemblyName => Assembly.Load(assemblyName));
+            if (ProcessedAssemblies.Contains(GetAssemblyKey(assembly)))
+            {
+                return;
+            }
 
-            JsiiAssemblyAttribute attribute = assembly.GetCustomAttribute<JsiiAssemblyAttribute>();
+            var attribute = assembly.GetCustomAttribute<JsiiAssemblyAttribute>();
             if (attribute == null)
             {
+                ProcessedAssemblies.Add(GetAssemblyKey(assembly));
```

Comment:
You can move this to line 32 so there's a single code path that adds to the map

Reply:
This map add is a short-circuit for non-jsii assemblies. It makes it so each assembly is reflected upon only once.